packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -126,52 +155,93 @@ const dialPadButtonsDefault: _DialpadButtonProps[][] = [
   [{ primaryContent: '*' }, { primaryContent: '0', secondaryContent: '+' }, { primaryContent: '#' }]
 ];
 
+/* @conditional-compile-remove(dialpad) */
+const DtmfTones: DtmfTone[] = [
+  'Num1',
+  'Num2',
+  'Num3',
+  'Num4',
+  'Num5',
+  'Num6',
+  'Num7',
+  'Num8',
+  'Num9',
+  'Star',
+  'Num0',
+  'Pound'
+];
+
+/* @conditional-compile-remove(dialpad) */
 const DialpadContainer = (props: {
-  errorText: string;
   defaultText: string;
-  dialpadButtons?: _DialpadButtonProps[][];
-  styles?: _DialpadStyles;
+  // dialpadButtons?: DialpadButtonProps[][];
+  onSendDtmfTone?: (dtmfTones: DtmfTone) => Promise<void>;
+  // Callback for dialpad button behavior
+  onClickDialpadButton?: () => void;
+  // customize dialpad input formatting
+  onDisplayDialpadInput?: (input: string) => string;
+  styles?: DialpadStyles;
 }): JSX.Element => {
   const theme = useTheme();
   const [textValue, setTextValue] = useState('');
   const [error, setError] = useState('');
-  const onClickDialpad = (input: string): void => {
-    setError('');
 
+  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput } = props;
+
+  const onClickDialpad = (input: string, index: number): void => {
+    setError('');
     setTextValue(textValue + input);
+    if (onSendDtmfTone) {
+      onSendDtmfTone(DtmfTones[index]);
+    }
+    if (onClickDialpadButton) {
+      onClickDialpadButton();
+    }
   };
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const setText = (e: any): void => {
-    // if text content includes non-valid dialpad input return an error
-    if (!`${e.target.value}`.match(/^[0-9\s*#+()-]*$/)) {
-      setError(
-        _formatString(props.errorText, { invalidCharacter: `${e.target.value.slice(e.target.value.length - 1)}` })
-      );
-    } else {
-      setError('');
-      setTextValue(e.target.value);
-    }
+    setError('');
+    setTextValue(e.target.value);
   };
 
-  const dialpadButtonsContent = props.dialpadButtons ?? dialPadButtonsDefault;
+  // const dialpadButtonsContent = props.dialpadButtons ?? dialPadButtonsDefault;
```

Comment:
remove commented out line?

Reply:
This is commented out for now since we decided to not let contoso customize dialpad content, if in the future there is a request for that, we can uncomment this, I will add a comment to explain this